es6-shim.js

```diff
@@ -1844,13 +1844,7 @@
 
     atanh: function atanh(value) {
       var x = Number(value);
-      if (Number.isNaN(x) || x < -1 || x > 1) {
-        return NaN;
-      }
-      if (x === -1) { return -Infinity; }
-      if (x === 1) { return Infinity; }
-      if (x === 0) { return x; }
-      return 0.5 * _log((1 + x) / (1 - x));
+      return Math.sign(x) * Math.log1p(2 * Math.abs(x) / (1 - Math.abs(x))) / 2;
```

Comment:
NaNs, zeroes, and infinities are edge cases. Preserving them makes it clear to future refactorers of these methods that they don't need to rely on things like `Math.sign` etc to handle this for them.

Also, we'll want to use cached versions of `Math.sign`, `Math.log1p`, and `Math.abs`, to avoid breaking if someone deletes them off of the `Math` object.